glib-macros/src/lib.rs

```diff
@@ -275,16 +323,31 @@ use utils::{parse_nested_meta_items_from_stream, NestedMetaItem};
 /// # }
 /// impl Foo {
 ///     fn foo(&self) {
-///         let closure = clone!(@strong self.v as v => move |x| {
-///             println!("self.v: {}", v);
-///         });
+///         let closure = clone!(
+///             #[strong(rename_to = v)]
+///             self.v,
+///             move |x| {
+///                 println!("self.v: {}", v);
+///             },
+///         );
 ///         # closure(2);
 ///     }
 /// }
 /// ```
 #[proc_macro]
 pub fn clone(item: TokenStream) -> TokenStream {
-    clone::clone_inner(item)
+    // Check if this is an old-style clone macro invocation.
+    // These always start with an '@' punctuation.
+    let Some(first) = item.clone().into_iter().next() else {
+        return syn::Error::new(Span::call_site(), "expected a closure or async block")
+            .to_compile_error()
+            .into();
+    };
+
+    match first {
+        TokenTree::Punct(ref p) if p.to_string() == "@" => clone_old::clone_inner(item),
```

Comment:
glib should get an `unstable-clone-syntax` feature flag, that maps to the same flag in `glib-macros`. If that flag is *not* enabled, then this and the other functions are directly mapping to `clone_old` / `closure_old` instead of doing anything else like they do now